packages/core/src/storage.ts

```diff
@@ -208,6 +208,68 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  searchTitle?: string
+  searchObjectId?: string
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export interface SearchQuery {
+  query: string
+  classes?: Ref<Class<Doc>>[]
+  spaces?: Ref<Space>[]
+}
+
+/**
+ * @public
+ */
+export interface SearchOptions {
+  limit?: number
+}
+
+/**
+ * @public
+ */
+export interface SearchResultDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  icon?: string
+  iconComponent?: string
+  iconProps?: { [key: string]: string }
+  objectId?: string
```

Comment:
What is difference between id and objectId?

Reply:
Must be really not a good naming, yes. objectId is "TSK-15" . Like human readable one.